_P101_TMP117.ino

```diff
@@ -0,0 +1,269 @@
+        success = true;
+        break;
+      }
+         
+   case PLUGIN_INIT:
+    {
+      Plugin_101_init = true;
+      success = true;
+      break;
+  }
+  
+    case PLUGIN_READ:
+      {
+ /*  command   parameter   description
+   *    0           X           print actual temperature
+   *    1           X           print EEPROM NIST UUID [E1|E2|E3]
+   *    2         float         set temperature offset like "2 20.5" sets the offset to 20.5°C
+   *    3         float         calibrate sensor to target temperature like "3 30.5" will calibrate sensor to 30.5°C
+   * 
+   */
+        // Get sensor resolution configuration
+         uint8_t ConvTime = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
+         uint8_t Averaging = Settings.TaskDevicePluginConfig[event->TaskIndex][1];
+         float adjust = Settings.TaskDevicePluginConfigFloat[event->TaskIndex][2];
+         float offset = Settings.TaskDevicePluginConfigFloat[event->TaskIndex][3];
+         
+        // Initiate wire library and serial communication
+  Wire.begin();           
+  // Serial.begin(115200);
+  /* The default setup is :
+   *    Conversion mode = CONTINUOUS  ---> continuoss
+   *    Conversion time = C125mS      -|
+   *    Averaging mode  = AVE8        -|-> new data every 125mS
+   *    Alert mode      = data        ---> alert pin states that new data is available
+   *    
+   */
+  tmp.init ( NULL );                // no callback
+  tmp.setConvMode (CONTINUOUS);     // contious measurement, also ONESHOT or SHUTWDOWN possible
+
+ switch (ConvTime)
+        {
+        case (0x00):
+          tmp.setConvTime (C15mS5);         // 1. setup C125mS+NOAVE = 15.5 mS measurement time
+          break;
+        case (0x01):
+          tmp.setConvTime (C125mS);         
+          break;
+        case (0x02):
+          tmp.setConvTime (C250mS);
+          break;
+        case (0x03):
+          tmp.setConvTime (C500mS);
+          break;
+        case (0x04):
+          tmp.setConvTime (C1S);
+          break;
+        case (0x05):
+          tmp.setConvTime (C4S);
+          break;
+        case (0x06):
+          tmp.setConvTime (C8S);
+          break;
+        case (0x07):
+          tmp.setConvTime (C16S);
+          break;   
+        }
+
+ switch (Averaging)
+        {
+        case (0x00):
+          tmp.setAveraging (NOAVE); 
+          break;
+        case (0x01):
+          tmp.setAveraging (AVE8);         
+          break;
+        case (0x02):
+          tmp.setAveraging (AVE32);
+          break;
+        case (0x03):
+          tmp.setAveraging (AVE64);
+          break;
+        }       
+
+          tmp.setOffsetTemperature ( offset );      //set temperature offset
+          tmp.setTargetTemperature ( adjust );      //calibrate sensor to target temperature 
+          tmp.update();
+
+            double temperature = tmp.getTemperature();
+            UserVar[event->BaseVarIndex] = temperature;

```

Comment:
Why use a double? You don't use it anywhere else in the code, other than assigning it to the `UserVar`, which is a float.

Use of `double` on the ESP8266 is done in software as the CPU cannot deal with double resolution.
So please do a direct assign instead of storing it in a variable.

Reply:
because tmp.getTemperature () only accepts double
can I use UserVar as a double?
I don't know how the plugin is structured, but from a dava or integer or float guide, I would need double
ok done
UserVar[event->BaseVarIndex] = TMP117_tmp.getTemperature();